### ISSUE TITLE
Adding the project to HACS

### ISSUE THREAD

**Issue by Arubinu** [BWilky/viewtimeout#4]
Hello, is it possible to add your project to HACS to simplify its installation and potentially make it known?

**Comment by jeremysherriff** [comment]
+1

With no instructions for installation in the ReadMe, it makes this difficult to use.
There is also confusion about dependencies (#1 and #8)

There are at least [two](https://community.home-assistant.io/t/subview-timout-auto-close-back/) [open](https://community.home-assistant.io/t/timeout-on-subview-new-feature-to-return-after-select-period/) feature requests for what this project already achieves.

@BWilky in the meantime can you update the ReadMe to include some steps for installation?  What files should be placed where, and does the resource need to be added to the HA Lovelace resource list?

**Comment by RuneMason1** [comment]
I agree on HACS but also some instructions. I don't know exactly what to put where. Do we need to put your JS file somewhere? Then the yaml on the main page in the dash

**Comment by jhol-byte** [comment]
I added "https://github.com/BWilky/viewtimeout.git" as user-defined repository in HACS and then I could install via HACS

**Comment by NR3Y** [comment]
Hmmm, didn't work for me. HACS gave me an error about a missing JSON file.

Anyway, I created a directory under /config/www called viewtimeout and dropped the timeout.js file into there. However, I feel that something should be added to configuration.yaml (or maybe not - just a gut feeling).

The resource has been added:

![Screenshot 2024-11-22 at 8 44 42 AM](https://github.com/user-attachments/assets/6cb3cf47-725c-47f8-9003-f46190326466)

My default dashboard view contains the folowing code

```
view_timeout:
  timeout_true: true
  default: home
  duration: 10000
  views:
    interior-lights: home
views:
```
When I navigate to the interior-lights page and wait 10 seconds, absolutely nothing happens.

Pointers as to what else I need to do?

**Comment by jhol-byte** [comment]
You need additionally lovelace-card-tool 

add in configuration.yaml I have:

```
frontend:
  extra_module_url:
      - /config/www/community/lovelace-card-tool
      - /config/www/community/viewtimeout 
```

**Comment by NR3Y** [comment]
Hmmm. still no change.

I put the viewtimeout folder (which has the timeout.js file in it) under config/www/community. It's referenced correctly in configuration.yaml

<img width="788" alt="Screenshot 2024-11-22 at 12 31 14 PM" src="https://github.com/user-attachments/assets/b3f4c6f9-22f5-49db-8d73-a0c47a0eeaec">

I also ammeneded the resource in the main view (three dots) to be:

<img width="1371" alt="Screenshot 2024-11-22 at 12 30 44 PM" src="https://github.com/user-attachments/assets/1f33b37d-e69e-455b-a52c-541e2d9a91f5">

So, everything looks OK, I think. If I go to the main page, then click on a link that should take me to a subview ("Interior Lights"), I navigate correctly, but it stays on that page - doesn't time out and go back to the main view (or anywhere for that matter).

**Comment by jhol-byte** [comment]
Are you sure that your paths "/local/community/..." are correct, shouldn't it be "/local/www/community/..." or similar? 


And it's "lovelace-card-tool" without trailing "s", my settings:

```
  extra_module_url:
    - /config/www/community/lovelace-card-tool
    - /config/www/community/viewtimeout 
```

My dashboard looks like this, be sure to really have a "path: home" :

```
view_timeout:
  timeout: true
  default: home
  duration: 60000
views:
  - type: sections
    path: home
    sections:
      - type: grid
```

Try to restart HA, clear browser cache, etc.

**Comment by NR3Y** [comment]
Well, I changed everything from /local/community to /config/www/community, just to be sure (including the resources). I also restarted HA and cleared the browser cache. Still no redirection.

Let me see if I got this correct:

- I have a main panel card (of View Type = Panel), which is my default home card. The URL for this panel is "home"
- I have a subview (on the same dashboard) for all the interior lights. (I'll have plenty more subviews, when I get this first one fixed.)
- If I click on an entity on my main panel (called "Interior Lights"), it does navigate to the subview correctly
- What I am trying to achieve is that the subview stays 'active' for about 10-seconds, then automatically it defaults back to the main panel.

Now, let's see if my configuration matches what I'm trying to achieve:

- I copied timeout.js into a new folder named viewtimeout
- That folder is located under the directory structure: /config/www/community
- configuration.yaml has the entries listed below:
```
frontend:
  extra_module_url:
    - /local/community/custom-brand-icons/custom-brand-icons.js
    - /local/card-mod.js
    - /config/www/community/lovelace-card-tools
    - /config/www/community/viewtimeout
 ```

- lovelace-card-tools matches what is in /config/www/community (yes, with a trailing 's')
- Under the three vertical dots of this main view, there is an added resource (/config/www/community/viewtimeout/timeout.js. Type Javascript).
- Under the three vertvail dots on the main view, using the raw editor, I have below:

```
view_timeout:
  timeout_true: true
  default: home
  duration: 10000
  views:
    interior-lights: home
views:
  - title: Main Panel
    type: panel
    path: home
    theme: ios-dark-mode-blue-red
    cards:
      - type: custom:layout-card
        layout_type: custom:grid
```
- The main view has the keyword: path: home
- The subview has the keyworfd: path: interior-lights
- Under the view_timeout: section, as shown above, I only want the timeout to be active on the interior-lights subview.

I have restarted HA and cleared the cache, but still no auto-navigation.

Should there be anything in the log file that would help narrow down the issue?

**Comment by NR3Y** [comment]
Still the same.

I renamed the stuff in configuration.yaml to /local/community/xxxx, including the .js filename:

<img width="489" alt="Screenshot 2024-11-23 at 1 00 52 PM" src="https://github.com/user-attachments/assets/57719a1f-099c-4b5c-9859-22d92c454ce7">

I also have the resources for lovelace-card-tools in there as well:


<img width="565" alt="Screenshot 2024-11-23 at 12 05 16 PM" src="https://github.com/user-attachments/assets/e879ca5e-21b1-4765-88f6-b00db4a911ee">

I wonder if installing via HACS is the answer, in that everything gets put in the same place with the right names? If you think that's a good idea, how do I do that? (I searched for viewtimeout under HACS and couldn't find it, I also tried adding a custom repository, but it threw out a missing JSON error.)

Also, thanks for your patience.
